presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1914,24 +1931,49 @@ private void analyzeOrderBy(Query node, Scope scope)
             analysis.setOrderByExpressions(node, orderByFieldsBuilder.build());
         }
 
-        private Scope createAndAssignScope(Node node, Scope parent, Field... fields)
+        private Scope createAndAssignScope(Node node, Optional<Scope> parentScope)
         {
-            return createAndAssignScope(node, parent, new RelationType(fields));
+            return createAndAssignScope(node, parentScope, emptyList());
         }
 
-        private Scope createAndAssignScope(Node node, Scope parent, List<Field> fields)
+        private Scope createAndAssignScope(Node node, Optional<Scope> parentScope, Field... fields)
         {
-            return createAndAssignScope(node, parent, new RelationType(fields));
+            return createAndAssignScope(node, parentScope, new RelationType(fields));
         }
 
-        private Scope createAndAssignScope(Node node, Scope parent, RelationType relationType)
+        private Scope createAndAssignScope(Node node, Optional<Scope> parentScope, List<Field> fields)
         {
-            Scope scope = Scope.builder()
-                    .withParent(parent)
-                    .withRelationType(relationType)
-                    .build();
+            return createAndAssignScope(node, parentScope, new RelationType(fields));
+        }
+
+        private Scope createAndAssignScope(Node node, Optional<Scope> parentScope, RelationType relationType)
+        {
+            Scope.Builder scopeBuilder = scopeBuilder(parentScope);
+            scopeBuilder.withRelationType(relationType);
+
+            Scope scope = scopeBuilder.build();
             analysis.setScope(node, scope);
             return scope;
         }
+
+        private Scope createScope(Optional<Scope> parentScope)
+        {
+            return scopeBuilder(parentScope).build();
+        }
+
+        private Scope.Builder scopeBuilder(Optional<Scope> parentScope)
+        {
+            Scope.Builder scopeBuilder = Scope.builder();
+
+            if (parentScope.isPresent()) {
+                scopeBuilder.withParent(parentScope.get());
```

Comment:
yes, parent scope (local query scope hierarchy) should already have outer scope as it's ancestor. Added comment and `checkState` in analyze method